src/communication/response_broker.hpp

```diff
@@ -179,8 +212,10 @@ namespace cubcomm
       std::lock_guard<std::mutex> lk_guard (m_mutex);
 
```

Comment:
```suggestion
assert (m_response_payloads.find (a_rsn) != m_response_payloads.cend ());
```
Nit. to make sure that it's registered in advance.

Reply:
This is kind of given. There can't be any other thread/entity which touches that response payload. If there is - the whole server should come crashing down right away. So the assert adds no further value.